File: Algorithm/Java/Programmers/Level1/문자열 내 마음대로 정렬하기.java
//https://programmers.co.kr/learn/courses/30/lessons/12915

class Solution {
    public String[] solution(String[] strings, int n) {


        String answer[] = new String[strings.length];

        for(int i=0; i<strings.length; i++) {

            char c1 = strings[i].charAt(n);
            int idx = i;
            String s1 = strings[i];

            for(int j=i+1; j<strings.length; j++) {

                char c2 = strings[j].charAt(n);
                String s2 = strings[j];


                if(c1 > c2) {
                    c1 = c2;
                    s1 = s2;
                    idx = j;
                } else if(c1 == c2 && s1.compareTo(s2) > 0) {
                    s1 = s2;
                    idx = j;
                }

            }//2중 for

            String temp = strings[i];
            strings[i] = strings[idx];
            strings[idx] = temp;

        }

        System.out.println("abcde".compareTo("abc"));

        for(int i=0; i<strings.length; i++)
            answer[i] = strings[i];

        return answer;
    }
}

File: JavaScript/Programmers/Level1/문자열 내 마음대로 정렬하기.js
function solution(strings, n) {
  var answer = strings.sort((a,b) => {
    let c1 = a.charAt(n);
    let c2 = b.charAt(n);
    if(c1 < c2)
      return -1;
    else if(c1 > c2)
      return 1;
    else if(c1 == c2) {
      if(a < b) {
        return -1;
      } else {
        return 1;
      }
    }
  });
  return answer;
}

/*
function solution(strings, n) {
    return strings.sort((s1, s2) => s1[n] === s2[n] ? s1.localeCompare(s2) : s1[n].localeCompare(s2[n]));
}
*/